### PR TITLE
Replaced package "pkg_resources" with "packaging"

### LINKED ISSUE #7559: `DeprecationWarning`: `pkg_resources` is deprecated as an API in `utils/module.py`
The following warning has appeared in our `pytest` output:

```python
=============================== warnings summary ===============================
python/envs/venv_sct/lib/python3.9/site-packages/monai/utils/module.py:396
  /home/runner/work/spinalcordtoolbox/spinalcordtoolbox/python/envs/venv_sct/lib/python3.9/site-packages/monai/utils/module.py:396:"""
DeprecationWarning: pkg_resources is deprecated as an API. 
See https://setuptools.pypa.io/en/latest/pkg_resources.html
    pkg = __import__(module)  # top level module
"""
```

MONAI version: `1.3.0`

Skimming the MONAI source code, it seems like MONAI made this an optional import in https://github.com/Project-MONAI/MONAI/pull/2482:

https://github.com/Project-MONAI/MONAI/blob/ec63e068fec85dff5f043d73f2c029b359332707/monai/utils/module.py#L591

But, it seems like `pkg_resources` shouldn't be used at all: https://setuptools.pypa.io/en/latest/pkg_resources.html

### PR DESCRIPTION
**Pull request by K-Rilla**
Fixes #7559 .

### Description  
Replaced "pkg_resources" references with "packaging" in MONAI/monai/utils/module.py & setup.py  
Changes were made in functions "pytorch_after", "version_leq", "version_geq".

### Types of changes
- Non-breaking change (fix or new feature that would not break existing functionality).

### CODE REVIEW

**Comment by KumoLiu** [comment]
please also include the packaging in the "requirements-min.txt" to avoid the test error. Thanks!
https://github.com/Project-MONAI/MONAI/blob/dev/requirements-min.txt

File: monai/utils/module.py [hunk 1]
```diff
@@ -27,9 +27,12 @@
 from re import match
 from types import FunctionType, ModuleType
 from typing import Any, Iterable, cast
-
 import torch
 
+import importlib.metadata
+from packaging import version
```
Comment (KumoLiu) [inline monai/utils/module.py:33]: please remove the import here since there is already `optional_import` below.

File: monai/utils/module.py [hunk 2]
```diff
@@ -564,11 +567,11 @@ def version_leq(lhs: str, rhs: str) -> bool:
     """
 
     lhs, rhs = str(lhs), str(rhs)
-    pkging, has_ver = optional_import("pkg_resources", name="packaging")
+    pkging, has_ver = optional_import("packaging", name="packaging")
```
Comment (KumoLiu) [inline monai/utils/module.py:570]: pkging is even not used here.
```suggestion
    pkging, has_ver = optional_import("packaging")
```

File: monai/utils/module.py [hunk 3]
```diff
@@ -591,11 +594,11 @@ def version_geq(lhs: str, rhs: str) -> bool:
 
     """
     lhs, rhs = str(lhs), str(rhs)
-    pkging, has_ver = optional_import("pkg_resources", name="packaging")
+    pkging, has_ver = optional_import("packaging", name="packaging")
```
Comment (KumoLiu) [inline monai/utils/module.py:597]:
```suggestion
    pkging, has_ver = optional_import("packaging")
```

File: monai/utils/module.py [hunk 4]
```diff
@@ -564,11 +567,11 @@ def version_leq(lhs: str, rhs: str) -> bool:
     """
 
     lhs, rhs = str(lhs), str(rhs)
-    pkging, has_ver = optional_import("pkg_resources", name="packaging")
+    pkging, has_ver = optional_import("packaging", name="packaging")
     if has_ver:
         try:
-            return cast(bool, pkging.version.Version(lhs) <= pkging.version.Version(rhs))
-        except pkging.version.InvalidVersion:
+            return cast(bool, version(lhs) <= version(rhs))
```
Comment (KumoLiu) [inline monai/utils/module.py:573]: After the change above, you can just revert the change here.

File: monai/utils/module.py [hunk 5]
```diff
@@ -591,11 +594,11 @@ def version_geq(lhs: str, rhs: str) -> bool:
...
     """
     lhs, rhs = str(lhs), str(rhs)
-    pkging, has_ver = optional_import("pkg_resources", name="packaging")
+    pkging, has_ver = optional_import("packaging", name="packaging")
     if has_ver:
         try:
-            return cast(bool, pkging.version.Version(lhs) >= pkging.version.Version(rhs))
-        except pkging.version.InvalidVersion:
+            return cast(bool, version(lhs) >= version(rhs))
+        except version.InvalidVersion:
```
Comment (KumoLiu) [inline monai/utils/module.py:601]: Same as above.

**Comment by KumoLiu** [comment]
Hi @K-Rilla, please take a look at the comments, I guess you misunderstand several points. Feel free ask if you met any issue. Thanks!

File: monai/utils/module.py (hunk 1 above)
Comment (K-Rilla) [reply monai/utils/module.py:33]: Hello there,
Removing line 32 "import importlib.metadata" works fine.
However, removing line 33 "from packaging import version" causes issues since there are references for 'version' in the script...
Please advise

File: monai/utils/module.py (hunk 2 above)
Comment (ken-ni) [reply monai/utils/module.py]: Hello,
We removed references to pkging and pkg_resources but optional_import returns a tuple. Do you want us to keep pkging in the optional_import or is there any other suggestions you had?

File: monai/utils/module.py (hunk 4 above)
Comment (ken-ni) [reply monai/utils/module.py:572]: Hello,
Can you please clarify what you mean by 'revert the change here'? We took out references to pkging and changed the code in line 573 as requested.

File: monai/utils/module.py [hunk 6]
```diff
@@ -564,11 +566,11 @@ def version_leq(lhs: str, rhs: str) -> bool:
     """
 
     lhs, rhs = str(lhs), str(rhs)
-    pkging, has_ver = optional_import("pkg_resources", name="packaging")
+    has_ver = optional_import("packaging")
```
Comment (KumoLiu) [inline monai/utils/module.py:569]: Please change the code like this, and remove the import above and try to use `pkging.version` below.
```suggestion
    pkging, has_ver = optional_import("pkg_resources")
```

File: monai/utils/module.py (hunk 4 above)
Comment (KumoLiu) [reply monai/utils/module.py:572]: After change what I recommand above, the change here can just be reverted. Since you can use `pkging.version.Version(lhs)` to get the version from the `lhs`.
```
>>> pkging, has_ver = optional_import("packaging")
>>> pkging.version.Version("1.3.1a")
<Version('1.3.1a0')>
```

File: monai/utils/module.py (hunk 1 above)
Comment (ericspod) [reply monai/utils/module.py:33]: I think the point here is that `version` gets imported elsewhere within the body of functions using `optional_import` so it doesn't need importation here. If it does need to be here for type checking reasons, you may need to follow a pattern [like this ](https://github.com/Project-MONAI/MONAI/blob/f1ef3e88f4bc594779c0c4188883f78bf6d1efff/monai/handlers/logfile_handler.py#L22) where the `TYPE_CHECKING` variable is used to guard the import statement.

Comment (ericspod) [reply monai/utils/module.py:33]: Another issue is that "version" is a bit problematic as a name since a number of argument variables in functions below are "version".

File: monai/utils/module.py [hunk 7]
```diff
@@ -591,11 +590,11 @@ def version_geq(lhs: str, rhs: str) -> bool:
 
     """
     lhs, rhs = str(lhs), str(rhs)
-    pkging, has_ver = optional_import("pkg_resources", name="packaging")
+    pkging, has_ver = optional_import("packaging")
     if has_ver:
         try:
             return cast(bool, pkging.version.Version(lhs) >= pkging.version.Version(rhs))
-        except pkging.version.InvalidVersion:
+        except pkging.version.Version.InvalidVersion:
```
Comment (KumoLiu) [inline monai/utils/module.py:597]:
```suggestion
        except pkging.version.InvalidVersion:
```

File: monai/utils/module.py [hunk 8]
```diff
@@ -564,11 +563,11 @@ def version_leq(lhs: str, rhs: str) -> bool:
     """
 
     lhs, rhs = str(lhs), str(rhs)
-    pkging, has_ver = optional_import("pkg_resources", name="packaging")
+    pkging, has_ver = optional_import("packaging")
     if has_ver:
         try:
             return cast(bool, pkging.version.Version(lhs) <= pkging.version.Version(rhs))
-        except pkging.version.InvalidVersion:
+        except pkging.version.Version.InvalidVersion:
```
Comment (KumoLiu) [inline monai/utils/module.py:570]:
```suggestion
        except pkging.version.InvalidVersion:
```

File: requirements-dev.txt [hunk 9]
```diff
@@ -59,3 +59,4 @@ nvidia-ml-py
 huggingface_hub
 pyamg>=5.0.0
 git+https://github.com/Project-MONAI/GenerativeModels.git@7428fce193771e9564f29b91d29e523dd1b6b4cd
+packaging
```
Comment (KumoLiu) [inline requirements-dev.txt:62]:
```suggestion
```
No need this package here, requirement-min already include it.
https://github.com/Project-MONAI/MONAI/blob/de41c6403f83d53bc85563a23912aff83e49d5f6/requirements-dev.txt#L2

Comment (KumoLiu) [reply requirements-dev.txt:62]: Hi @K-Rilla, could you please remove the packaging here, it's duplicated. Thanks!

File: monai/utils/module.py [hunk 10]
```diff
@@ -591,7 +591,7 @@ def version_geq(lhs: str, rhs: str) -> bool:
 
     """
     lhs, rhs = str(lhs), str(rhs)
-    pkging, has_ver = optional_import("pkg_resources", name="packaging")
+    pkging, has_ver = optional_import("packaging.Version")
```
Comment (KumoLiu) [inline monai/utils/module.py:594]: For this one, did you try it works locally? 
```
>>> from packaging import Version
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: cannot import name 'Version' from 'packaging' (/usr/local/lib/python3.10/dist-packages/packaging/__init__.py)
```

**Comment by ericspod** [comment]
From the CI issue with building the wheel, `packaging` may be needed in places like [here](https://github.com/Project-MONAI/MONAI/blob/49a1e343129909f7e01c67882a71c12beed3ef30/.github/workflows/pythonapp.yml#L154) where setup-related things are installed with pip.

File: monai/utils/module.py (hunk 10 above)
Comment (ken-ni) [reply monai/utils/module.py:594]: Yes, we tried and this change works locally on our machines.

**Review by KumoLiu — APPROVED** [review]
